build-npm/build.sh

```diff
@@ -0,0 +1,346 @@
+    exit 1
+  fi
+
+  echo "DEBUG: Removing existing JFrog config..."
+  jf config remove repox > /dev/null 2>&1 # Do not log if the repox config were not present
+
+  echo "DEBUG: Adding JFrog config..."
+  if ! jf config add repox --artifactory-url "$ARTIFACTORY_URL" --access-token "$ARTIFACTORY_DEPLOY_ACCESS_TOKEN"; then
+    echo "ERROR: Failed to add JFrog config" >&2
+    exit 1
+  fi
+
+  echo "DEBUG: Configuring NPM repositories..."
+  if ! jf npm-config --repo-resolve "npm" --repo-deploy "$ARTIFACTORY_DEPLOY_REPO"; then
+    echo "ERROR: Failed to configure NPM repositories" >&2
+    exit 1
+  fi
+
+  echo "DEBUG: Publishing NPM package..."
+  if ! jf npm publish --build-name="$PROJECT" --build-number="$BUILD_NUMBER"; then
+    echo "ERROR: Failed to publish NPM package" >&2
+    exit 1
+  fi
+
+  jf rt build-collect-env "$PROJECT" "$BUILD_NUMBER"
+
+  echo "DEBUG: Publishing build info..."
+  local build_publish_output
+  if ! build_publish_output=$(jf rt build-publish "$PROJECT" "$BUILD_NUMBER"); then
+    echo "ERROR: Failed to publish build info" >&2
+    exit 1
+  fi
+
+  echo "DEBUG: Build publish output: ${build_publish_output}"
+
+  # Extract build info URL
+  local build_info_url
+  build_info_url=$(echo "$build_publish_output" | jq -r '.buildInfoUiUrl // empty')
+  if [ -n "$build_info_url" ]; then
+    echo "build-info-url=$build_info_url" >> "$GITHUB_OUTPUT"
+    echo "DEBUG: Build info URL saved: $build_info_url"
+  fi
+
+  echo "DEBUG: JFrog operations completed successfully"
+}
+
+# Common build steps to avoid repetition
+run_npm_install() {
+  echo "Installing npm dependencies..."
+  npm ci
+}
+
+run_npm_tests() {
+  if [ "$SKIP_TESTS" != "true" ]; then
+    echo "Running tests..."
+    npm test
+  else
+    echo "Skipping tests (SKIP_TESTS=true)"
+  fi
+}
+
+run_npm_build() {
+  echo "Building project..."
+  npm run build
+}
+
+# Complete build pipeline with optional steps
+# Usage: run_standard_pipeline <enable_sonar> <enable_deploy> [sonar_args...]
+run_standard_pipeline() {
+  local enable_sonar="${1:-true}"
+  local enable_deploy="${2:-true}"
+  shift 2  # Remove first two parameters
+  local sonar_args=("$@")  # Remaining parameters are sonar args
+
+  run_npm_install
+  run_npm_tests
+
+  if [ "${enable_sonar}" = "true" ]; then
+    run_sonar_scanner "${sonar_args[@]}"
+  fi
+
+  run_npm_build
+
+  if [ "${enable_deploy}" = "true" ]; then
+    jfrog_npm_publish
+  fi
+}
+
+build_npm() {
+  echo "=== NPM Build, Deploy, and Analyze ==="
+  echo "Branch: ${GITHUB_REF_NAME}"
+  echo "Pull Request: ${PULL_REQUEST}"
+  echo "Deploy Pull Request: ${DEPLOY_PULL_REQUEST}"
+  echo "Skip Tests: ${SKIP_TESTS}"
+
+  if is_main_branch && ! is_pull_request; then
+    echo "======= Building main branch ======="
+
+    local current_version
+    current_version=$(get_current_version)
+    echo "Current version: ${current_version}"
```

Comment:
See set_maven_build_version in https://github.com/SonarSource/ci-github-actions/blob/15c69114fa0c3ba1aa88321c3dc8405fbcc35bf2/build-maven/build.sh#L86

```
  local current_version
  set_project_version
```

It is possible to 
- merge get_current_version and set_npm_version_with_build_id (-> set_project_version for consistency?)
- call it only once at the beginning of build_npm
- don't use function output to pass values
- remove useless "${BUILD_NUMBER}" parameter
- remove useless check_version_format function